-- [E190] Potential Issue Warning: tests/neg-custom-args/captures/real-try.scala:38:2 ----------------------------------
38 |  b
   |  ^
   |  Discarded non-Unit value of type Cell[() -> Unit]. Add `: Unit` to discard silently.
   |
   | longer explanation available when compiling with `-explain`
-- Error: tests/neg-custom-args/captures/real-try.scala:14:2 -----------------------------------------------------------
14 |  try  // error
   |  ^
   |  The result of `try` cannot have type () => Unit since
   |  that type captures the root capability `cap`.
   |  This is often caused by a locally generated exception capability leaking as part of its result.
   |
   |  where:    => refers to a fresh root capability classified as Control in the type of given instance canThrow$1
15 |    () => foo(1)
16 |  catch
17 |    case _: Ex1 => ???
18 |    case _: Ex2 => ???
-- [E007] Type Mismatch Error: tests/neg-custom-args/captures/real-try.scala:21:4 --------------------------------------
21 |    () => foo(1) // error
   |    ^^^^^^^^^^^^
   |    Found:    () ->{canThrow$2} Unit
   |    Required: () ->'s1 Unit
   |
   |    Note that capability canThrow$2, defined in method try$2
   |    cannot be included in outer capture set 's1 of value x.
   |
   | longer explanation available when compiling with `-explain`
-- [E007] Type Mismatch Error: tests/neg-custom-args/captures/real-try.scala:27:4 --------------------------------------
27 |    () => Cell(foo(1)) // error
   |    ^^^^^^^^^^^^^^^^^^
   |    Found:    () ->{canThrow$3} Cell[Unit]^'s2
   |    Required: () ->'s3 Cell[Unit]^'s4
   |
   |    Note that capability canThrow$3, defined in method try$3
   |    cannot be included in outer capture set 's3 of value y.
   |
   | longer explanation available when compiling with `-explain`
-- [E007] Type Mismatch Error: tests/neg-custom-args/captures/real-try.scala:33:8 --------------------------------------
33 |    Cell(() => foo(1)) // error
   |    ^^^^^^^^^^^^^^^^^^
   |    Found:    Cell[() ->{canThrow$4} Unit]
   |    Required: Cell[() ->'s5 Unit]^'s6
   |
   |    Note that capability canThrow$4, defined in method try$4
   |    cannot be included in outer capture set 's5 of value b.
   |
   | longer explanation available when compiling with `-explain`
